iree/compiler/Dialect/Flow/Transforms/LegalizeInputTypes.cpp

```diff
@@ -85,9 +85,9 @@ LogicalResult convertOperation(Operation *oldOp,
                                FlowTypeConverter &typeConverter,
                                BlockAndValueMapping &mapping,
                                OpBuilder &builder) {
-  if (llvm::isa<mlir::linalg::LinalgOp>(oldOp)) {
-    // Currently assumes linalg ops have legal types.
-    // TODO: rewrite to generic and back.
+  if (isa<linalg::LinalgDialect>(oldOp->getDialect())) {
+    // Currently we assume all Linalg ops only contain valid types.
+    // TODO: Support converting Linalg types when unsupported.
     builder.clone(*oldOp, mapping);
     return success();
   }
```

Comment:
I personally think we should have a test for this in legalize_input_types.mlir... It looks like you are trying to allow linalg.yield, but it should only exist in Linalg bodys?

Reply:
Correct however this pass recurses into linalg bodies (probably shouldn't but it does). There are also other linalg ops that will likely appear so we should mark all of linalg as a special case until it is resolved. Test added.